marshmallow/schema.py

```diff
@@ -583,13 +586,17 @@ def _do_load(self, data, many=None, partial=None, postprocess=True):
         :param data: The data to deserialize.
         :param bool many: Whether to deserialize `data` as a collection. If `None`, the
             value for `self.many` is used.
-        :param bool partial: Whether to ignore missing fields. If `None`, the
-            value for `self.partial` is used.
+        :param bool|tuple partial: Whether to ignore missing fields. If `None`,
+            the value for `self.partial` is used. If its value is an iterable,
+            only missing fields listed in that iterable will be ignored.
         :param bool postprocess: Whether to run post_load methods..
         :return: A tuple of the form (`data`, `errors`)
         """
         many = self.many if many is None else bool(many)
-        partial = self.partial if partial is None else bool(partial)
+        if partial is None:
+            partial = self.partial
+        elif not utils.is_collection(partial):
```

Comment:
With `partial = partial or self.partial` if you pass `False`, `tuple()` or anything else that gets implicitly evaluated false, you'll end up using `self.partial`. So you need to explicitly check for the default value (None).

For example:
```python
s = Schema(partial=True)
s.load({}, partial=False)  # partial == True
```
We can drop the branch if it's OK to not coerce things into bool anymore though.